tchannel-core/src/main/java/com/uber/tchannel/channels/Connection.java

```diff
@@ -56,64 +58,77 @@ public Connection(Peer peer, @NotNull Channel channel, Direction direction) {
         if (channel.isActive() && this.state == ConnectionState.UNCONNECTED) {
             this.state = ConnectionState.CONNECTED;
         }
+        lock = new Object();
```

Comment:
Any reason to introduce an explicit object for locking - instead of locking on the instance itself?

Reply:
Connection is meant to be public. If someone syncs on it - it will break. If class extends it and syncs on this - it will break.
Connection should be unconditionally threadsafe as defined in Effective Java by Joshua Bloch Item70: Document thread safety. Also see the commit message